tests/qos/test_qos_sai.py

```diff
@@ -0,0 +1,461 @@
+"""
+    QoS SAI feature in SONiC.
+
+    Parameters:
+        --ptf_portmap <filename> (str): file name of port index to DUT interface alias map. Default is None.
+            In case of a filename is not provided, a file containing port indeces to aliases map will be generated.
+
+        --disable_test (bool): Disables experimental QoS SAI test cases. Deafult is True
+
+        --qos_swap_syncd (bool): Used to install the RPC syncd image before running the tests. Default is True.
+
+        --qos_dst_ports (list) Indeces of available DUT test ports to serve as destination ports. Note, This is not port
+            index on DUT, rather an index into filtered (excludes lag member ports) DUT ports. Plan is to randomize port
+            selection. Default is [0, 1, 3]
+
+        --qos_src_ports (list) Indeces of available DUT test ports to serve as source port. Similart note as in
+            qos_dst_ports applies. Default is [2]
```

Comment:
as specified above. get the indices from the constructed port list. In some of the tests, there is a requirement that the traffic should land on a single XPE.

Reply:
It does land on specific port Neetha. the file generated is identical to the one provided. I prefer to keep this was unless there is a need for the file which I did not see so far.